### DISCUSSION TITLE
Connecting to Planetscale in Spring Boot

### DISCUSSION THREAD

**Discussion by wardstoneX** [planetscale/discussion#208] [FAQs]
I'm developing a web application in Spring Boot,where i need to establish connection to a database. What i want to do is modify the apllication.properties file like writing the connection source, username etc.However, this didnt quite worked out for me. I could establish the connection via the connection String. However, this isnt suitable. Is there a way for me to achieve what i want?
Ps: i am new to these things. So please be understanding. Sorry for wrong category

**Comment by fooforge** [comment]
Hey @wardstoneX,

I'm not super well versed with Spring Boot, personally, but I can link you to two examples at least:

1. https://github.com/planetscale/examples/blob/main/java - This is PlanetScale's minimal example for Java.
2. https://github.com/spring-projects/spring-petclinic - This is a sample Spring Boot app with different database connection examples. See their [application-mysql.properties](https://github.com/spring-projects/spring-petclinic/blob/main/src/main/resources/application-mysql.properties) for a MySQL connection example

If none of these examples have helped, it would be great if you could provide some sample code and describe in greater detail what you're trying to achieve.

**Reply by jgreet** [reply]
Just to expand a little on @fooforge's answer, I was able to connect the [spring-petclinic](https://github.com/spring-projects/spring-petclinic) example by starting it like this:

```
MYSQL_URL=jdbc:mysql://3o1lmn41ewl7.us-east-1.psdb.cloud/spring?sslMode=VERIFY_IDENTITY MYSQL_USER=3f22xw62nqd1 MYSQL_PASS=pscale_pw_<redacted> mvn spring-boot:run -Dspring-boot.run.profiles=mysql
```

If that was a real application you'd probably want to hard code the database flavor in the properties file.

The JDBC URL can be copied from the [Connect dialog](https://docs.planetscale.com/tutorials/connect-any-application) under Java.

This example app won't actually run against PlanetScale because its schema requires foreign key constraints, but you can see that error being passed up from the database which at least verifies that the app has connected:

```java
Caused by: com.mysql.cj.jdbc.exceptions.MySQLQueryInterruptedException: foreign key constraints are not allowed, see https://vitess.io/blog/2021-06-15-online-ddl-why-no-fk/
```
